### PR TITLE
feat: Add spin-config import to the rust sdk

### PR DESCRIPTION
**Pull request by fibonacci1729**
This PR exposes the spin-config interface through the Rust SDK.
An example (similar to config-tinygo) is also included.

Signed-off-by: Brian H <brian.hardock@fermyon.com>

### CODE REVIEW

File: examples/config-rust/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build]
```
Comment (lann) [inline examples/config-rust/.cargo/config.toml:1]: Do we need another example for this? Can it be rolled into an existing one?

File: sdk/rust/src/lib.rs [hunk 2]
```diff
@@ -45,3 +45,26 @@ pub mod redis {
     /// Exports the generated outbound Redis items.
     pub use outbound_redis::*;
 }
+
+#[allow(missing_docs)]
```
Comment (lann) [inline sdk/rust/src/lib.rs:49]: :neutral_face:

[the remark this answers is not in the document]
Comment (fibonacci1729) [reply sdk/rust/src/lib.rs:49]: you betcha!

File: examples/config-rust/.cargo/config.toml (hunk 1 above)
Comment (fibonacci1729) [reply examples/config-rust/.cargo/config.toml:1]: Probably not, just following convention. I can roll this into `http-rust`

File: sdk/rust/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fibonacci1729) [reply sdk/rust/src/lib.rs:49]: Unfortunately, this doesn't appear to be possible so i'm keep as is.

[the remark this answers is not in the document]
Comment (lann) [reply sdk/rust/src/lib.rs:49]: What's it doing right above here at line 46?

**Review by lann — APPROVED** [review]

**Pull request merged**